serverlessworkflow/modules/ROOT/pages/security/authention-support-for-openapi-services.adoc

```diff
@@ -1 +1,370 @@
+
+|===
+
+The API Key scheme has an additional property `name` that specifies where to add the API Key in the request token: _header_, _cookie_ or _query_.
+However, it is automatically managed and no additional configurations are required. Following the above example the API Key will be passed as a http request parameter `api-key-name`.
+
+
+== OAuth2 Authentication
+
+The following example shows a `security-example.json` file that defines a `sayHelloOauth2` operation secured with the `oauth-example` scheme, and the supported configurations for that scheme type.
+
+[source, json]
+----
+{
+  "openapi": "3.1.0",
+  "info": {
+    "title": "Oauth2 Scheme Example",
+    "version": "1.0"
+  },
+  "paths": {
+    "/hello-with-oauth2": {
+      "get": {
+        "operationId": "sayHelloOauth2",
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "text/plain": {
+                "schema": {
+                  "type": "string"
+                }
+              }
+            }
+          }
+        },
+        "security": [{"oauth-example" : []}]
+      }
+    }
+  },
+  "components": {
+    "securitySchemes": {
+      "oauth-example": {
+        "type": "oauth2",
+        "flows": {
+          "clientCredentials": {
+            "authorizationUrl": "https://example.com/oauth",
+            "tokenUrl": "https://example.com/oauth/token",
+            "scopes": {
+            }
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+**These are the supported configurations:**
+
+Unlike the `http basic`, `http bearer` and `apiKey` security schemes, the Oauth2 authentication is relied on the https://quarkus.io/guides/security-openid-connect-client[Quarkus OpenId Connect and Oauth2 Clients and Filters].
+
+By using a simple naming convention, the token management operations will be delegated to a Quarkus `OidcClient`.
+
+For the example above, the token management operations for accessing the `sayHelloOauth2` operation will be delegated to the OidcClient `oauth_example`.
+
+You can configure this `OidcClient` as:
+
+[source, properties]
+----
+quarkus.oidc-client.oauth_example.auth-server-url=https://example.com/oauth
+quarkus.oidc-client.oauth_example.token-path=/tokens
+quarkus.oidc-client.oauth_example.discovery-enabled=false
+quarkus.oidc-client.oauth_example.client-id=kogito-app
+quarkus.oidc-client.oauth_example.grant.type=client
+quarkus.oidc-client.oauth_example.credentials.client-secret.method=basic
+quarkus.oidc-client.oauth_example.credentials.client-secret.value=secret
+----
+
+The configuration suffix `quarkus.oidc-client.oauth_example` is exclusive for the schema defined in the specification file and the scheme name is sanitized by applying the rules described <<sanitize_note, above>>.
+
+For this to work you **must** add the https://quarkus.io/guides/security-openid-connect-client#oidc-client-filter[Quarkus OIDC Client Filter Extension] to your project:
+
+[source, xml]
+----
```

Comment:
A tip was added here to refresh the reader memory and help him to use the Quarkus CLI too.
![Screenshot from 2022-06-14 15-42-29](https://user-images.githubusercontent.com/2431454/173592341-39c31803-dc5a-4f65-94ba-23ad1493f5b1.png)

Reply:
well, it depends...., I believe this guide is more a kind of documentation guide rather than a step-by-step.one. A reader that reach this guide has normally already went through step-by-step guides, like the "create a hello world SW", and is a developer that will directly copy paste the dependency. I think It's ok.